docker-compose.yml

```diff
@@ -10,6 +10,7 @@ services:
       - ./plugins/events:/var/www/html/admin/tool/events
       - ./plugins/study:/var/www/html/mod/study
       - ./plugins/content:/var/www/html/filter/content
+      - ./plugins/direct_events_plugin:/var/www/html/local/direct
       - .:/repo
     networks:
```

Comment:
Another comment I'm making here for lack of a better place: Just to get us into the habit of using reasonable code practices with Moodle (e.g. things we'll have to have pass in any real plugin), please review / address any issues from running the `Code checker` and `Moodle PHPdoc check` which you can find in the `Development` tab of the admin page in local environments. Also a couple of local links for convenience:

* http://localhost:8000/local/codechecker/index.php
* http://localhost:8000/local/moodlecheck/index.php

The path input both of the :point_up: ask for is relative to the Moodle code base (e.g. prior to addressing my naming comments above you'd put `local/direct`.

@P-Gill97 - FYI as :point_up: will be useful for you as well in k12-182

Reply:
Nudging this comment from the first pass.

Reply:
I'd worked on these the first time around, and looking at them again I'm running into similar issues. Certain lines the code checker complains about have identical structures to what they're asking for...🤔 maybe you've run into these problems before @rnathuji ?

I'm also not sure where to start with phpdocs. Maybe we could look at this after lunch?

Reply:
Sure -- you can also reference [upstream Moodle code](https://github.com/moodle/moodle), [plugins from Catalyst](https://github.com/catalyst), and our [moodle-local_raisecli](https://github.com/openstax/moodle-local_raisecli) plugin for examples. I suspect we're missing things like:

* The [license snippet](https://github.com/openstax/moodle-local_raisecli/blob/e7aadc577c5469b295c511b70e2e54cb435d1bef/version.php#L2-L15)
* PHP doc strings like [this one](https://github.com/openstax/moodle-local_raisecli/blob/e7aadc577c5469b295c511b70e2e54cb435d1bef/externallib.php#L50-L56). In the past I mostly cobble these together based upon examples.